tiledb/sm/query/sparse_unordered_with_dups_reader.cc

```diff
@@ -454,70 +460,79 @@ Status SparseUnorderedWithDupsReader::create_result_tiles(bool* tiles_found) {
     bool budget_exceeded = false;
     unsigned int f = 0;
     while (f < fragment_num && !budget_exceeded) {
-      auto range_it = result_tile_ranges_[f].rbegin();
-      uint64_t t = 0;
-      while (range_it != result_tile_ranges_[f].rend()) {
-        for (t = range_it->first; t <= range_it->second; t++) {
+      if (!all_tiles_loaded_[f]) {
+        auto range_it = result_tile_ranges_[f].rbegin();
+        while (range_it != result_tile_ranges_[f].rend()) {
+          auto last_t = result_tile_ranges_[f].front().second;
+
           // Figure out the start index.
           auto start = range_it->first;
           if (!result_tiles_.empty() && result_tiles_.back().frag_idx() == f) {
             start = std::max(start, result_tiles_.back().tile_idx() + 1);
           }
 
-          RETURN_NOT_OK(add_result_tile(
-              dim_num,
-              memory_budget_result_tiles,
-              memory_budget_qc_tiles,
-              memory_budget_coords,
-              f,
-              t,
-              domain,
-              &budget_exceeded));
-          *tiles_found = true;
+          for (uint64_t t = start; t <= range_it->second; t++) {
+            RETURN_NOT_OK(add_result_tile(
+                dim_num,
+                memory_budget_result_tiles,
+                memory_budget_qc_tiles,
+                memory_budget_coords,
+                f,
+                t,
+                last_t,
+                domain,
+                &budget_exceeded));
+
+            if (budget_exceeded) {
+              if (result_tiles_.empty())
+                return LOG_STATUS(Status::SparseUnorderedWithDupsReaderError(
+                    "Cannot load a single tile, increase memory budget"));
+              break;
+            }
+          }
 
           if (budget_exceeded)
             break;
+          range_it++;
         }
-
-        if (budget_exceeded)
-          break;
-        range_it++;
       }
 
-      all_tiles_loaded_[f] = !budget_exceeded;
       f++;
     }
   } else {
     // Load as many tiles as the memory budget allows.
     bool budget_exceeded = false;
     unsigned int f = 0;
     while (f < fragment_num && !budget_exceeded) {
-      uint64_t t = 0;
-      auto tile_num = fragment_metadata_[f]->tile_num();
+      if (!all_tiles_loaded_[f]) {
```

Comment:
(as noted in 'if' branch above)
set elsewhere than old line 520 below (now gone),
or irrelevant?

Reply:
It's now set in SparseUnorderedWithDupsReader::add_result_tile, line 393.